File: C#/0637. Average of Levels in Binary Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public IList<double> AverageOfLevels(TreeNode root) {
        IList<double> rep=new List<double>();
        if(root==null){
            return rep;
        }
        Queue<TreeNode> rootList=new Queue<TreeNode>();
        rootList.Enqueue(root);
        while(rootList.Count!=0){
            List<int> tmp=new List<int>();
            int lenOfRootList=rootList.Count;
            for(int i=0;i<lenOfRootList;i++){
                TreeNode curRoot=rootList.Dequeue();
                tmp.Add(curRoot.val);
                if(curRoot.left!=null){
                    rootList.Enqueue(curRoot.left);
                }
                if(curRoot.right!=null){
                    rootList.Enqueue(curRoot.right);
                }
            }
            rep.Add((double)tmp.Average());
        }
        return rep;
    }
}

File: archives/20190519python/0637. Average of Levels in Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def averageOfLevels(self, root: TreeNode) -> List[float]:
        rep=[]
        if not root:
            return rep
        
        root_list=[root]
        while root_list:
            len_root_list=len(root_list)
            cur_sum=0
            for k in range(len_root_list):
                cur_root=root_list.pop(0)
                cur_sum+=cur_root.val
                if cur_root.left:
                    root_list.append(cur_root.left)
                if cur_root.right:
                    root_list.append(cur_root.right)
            rep.append(cur_sum/len_root_list)
        return rep
